libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -545,5 +566,11 @@ bool AC_Fence::geofence_failed() const
     if (get_breaches() != 0) {
         return true;
     }
+    if (_enabled_fences & AC_FENCE_TYPE_POLYGON) {
+        if (!_boundary_valid) {
+            return true;
```

Comment:
I don't like this as it is contrary to what the breach code does: if the boundary isn't valid, then it says that not breach occurred - so why would we report a failed geofence?

By the way, the term geofence here was badly applied, we call it polygon fence everywhere except these new methods.